test/internal/helpers/certificate.go

```diff
@@ -0,0 +1,76 @@
+package helpers
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+)
+
+// GenerateSelfSignedCert generates a tls.Certificate struct to be used in TLS client/listener configurations.
+func GenerateSelfSignedCert(commonName string, dnsNames []string) (tls.Certificate, error) {
```

Comment:
Given that we'll be using this for tests (because what else would use self signed certs?) how about we pass `t *testing.T` as first argument and not return an `error` but use the `t` in `require.NoError(t, err)` calls here?

This will also make it easier to call in tests and it will allow us to declare and assign in the same statement. WDYT?

Reply:
Secondly: how about adding options for commonName and dnsNames so that users can call this as

```
GenerateSelfSignedCert()
GenerateSelfSignedCert(WithCommonName("mycommonname"))
GenerateSelfSignedCert(WithCommonName("name"), WithDNSNames("dns1", "dns2"))
```

WDYT?

Reply:
> Secondly: how about adding options for commonName and dnsNames so that users can call this as
> 
> ```
> GenerateSelfSignedCert()
> GenerateSelfSignedCert(WithCommonName("mycommonname"))
> GenerateSelfSignedCert(WithCommonName("name"), WithDNSNames("dns1", "dns2"))
> ```
> 
> WDYT?

Sounds good. I think if we take this suggestion, it should be moved to a single package to call by `certificate.WithCommonName`, which would be much clearer then `helpers.WithCommonName`.

Reply:
Good idea! You can put it in https://github.com/Kong/kubernetes-ingress-controller/tree/main/test/helpers because using e.g. https://github.com/Kong/kubernetes-ingress-controller/tree/main/test/internal/helpers would prevent it from being used in unit tests

Reply:
So why do we have `test/helpers` and `test/internal/helpers`? What does the "internal" mean here, and what is the difference between the two paths?

Reply:
done.

Reply:
The internal package was added with the intention that it couldn't be used outside of KIC's module but because of the unfortunate placement under `test/internal` it ended up impossible to be used by unit tests in e.g. `pkg/`. We might reorganize this at some point 😅